.golangci.toml

```diff
@@ -16,21 +16,44 @@ confidence = 3
 [linters-settings.depguard.rules.main]
 allow = [] # allow all
 deny = [
-  {pkg = "io/ioutil", desc = "Deprecated: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details."},
-  {pkg = "gopkg.in/yaml.v2", desc = "Grafana packages are not allowed to depend on gopkg.in/yaml.v2 as gopkg.in/yaml.v3 is now available"},
-  {pkg = "github.com/pkg/errors", desc = "Deprecated: Go 1.13 supports the functionality provided by pkg/errors in the standard library."},
-  {pkg = "github.com/xorcare/pointer", desc = "Use pkg/util.Pointer instead, which is a generic one-liner alternative"},
-  {pkg = "github.com/gofrs/uuid", desc = "Use github.com/google/uuid instead, which we already depend on."},
+  { pkg = "io/ioutil", desc = "Deprecated: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details." },
+  { pkg = "gopkg.in/yaml.v2", desc = "Grafana packages are not allowed to depend on gopkg.in/yaml.v2 as gopkg.in/yaml.v3 is now available" },
+  { pkg = "github.com/pkg/errors", desc = "Deprecated: Go 1.13 supports the functionality provided by pkg/errors in the standard library." },
+  { pkg = "github.com/xorcare/pointer", desc = "Use pkg/util.Pointer instead, which is a generic one-liner alternative" },
+  { pkg = "github.com/gofrs/uuid", desc = "Use github.com/google/uuid instead, which we already depend on." },
 ]
 
 [linters-settings.depguard.rules.coreplugins]
 deny = [
-  {pkg = "github.com/grafana/grafana/pkg/", desc = "Core plugins are not allowed to depend on Grafana core packages"},
-]
-files = [
-  "**/pkg/tsdb/testdatasource/*",
-  "**/pkg/tsdb/testdatasource/**/*",
+  { pkg = "github.com/grafana/grafana/pkg/api", desc = "Core plugins are not allowed to depend on Grafana core packages" },
```

Comment:
This looks like it could be hard to maintain - do we need to add to this list every time we add a folder under `pkg`? Is it feasible to list only the allowed dependencies?

Reply:
Yes, that's one of the cons of this approach.

> Is it feasible to list only the allowed dependencies?

Unfortunately no, the [deepguard](https://github.com/OpenPeeDeeP/depguard) implementation is a bit simplistic and if `allowed` is defined, it doesn't supersedes the `deny` list. If we define only `allow` rules, we'd need to add any possible package a plugin may import (which is basically the whole go.mod :sweat:)

Reply:
I guess if we moved core backend datasources outside of pkg (to the root) it would simplify things?

Reply:
yes, I guess we have three options:

 - Keep plugins as they are, with this PR as adaptation
 - Move only decoupled plugins to a different folder
 - Move all plugins to a different folder

The less friction approach would be this one but happy to hear other thoughts. What would you prefer?

Reply:
This could work as a start, but we might need to revisit it

Reply:
okay, let's go ahead then we can modify it in the future.